doc/hardware.rst

```diff
@@ -0,0 +1,426 @@
+
+If you have no kernel modules listed and nothing is showing using ``dmesg``
+then this implies the kernel I2C driver is not loaded. Enable the I2C as
+follows:
+
+#. Run ``sudo raspi-config``
+#. Use the down arrow to select ``5 Interfacing Options``
+#. Arrow down to ``P5 I2C``
+#. Select **yes** when it asks you to enable I2C
+#. Also select **yes** when it asks about automatically loading the kernel module
+#. Use the right arrow to select the **<Finish>** button
+
+After rebooting re-check that the ``dmesg | grep i2c`` command shows whether
+I2C driver is loaded before proceeding. You can also
+`enable I2C manually <http://elinux.org/RPiconfig#Device_Tree>`_ if the
+``raspi-config`` utility is not available.
+
+Optionally, to improve performance, increase the I2C baudrate from the default
+of 100KHz to 400KHz by altering ``/boot/config.txt`` to include::
+
+  dtparam=i2c_arm=on,i2c_baudrate=400000
+
+Then reboot.
+
+Next, add your user to the *i2c* group and install ``i2c-tools``::
+
+  $ sudo usermod -a -G i2c pi
+  $ sudo apt-get install i2c-tools
+
+Logout and in again so that the group membership permissions take effect, and
+then check that the device is communicating properly (if using a rev.1 board,
+use 0 for the bus, not 1)::
+
+  $ i2cdetect -y 1
+         0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
+    00:          -- -- -- -- -- -- -- -- -- -- -- -- --
+    10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
+    20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
+    30: -- -- -- -- -- -- -- -- -- -- -- UU 3c -- -- --
+    40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
+    50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
+    60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
+    70: -- -- -- -- -- -- -- --
+
+According to the man-page, "UU" means that probing was skipped, because the
+address was in use by a driver. It suggest that there is a chip at that
+address. Indeed the documentation for the device indicates it uses two
+addresses.
+
+SPI
+"""
+The GPIO pins used for this SPI connection are the same for all versions of the
+Raspberry Pi, up to and including the Raspberry Pi 3 B.
+
+.. warning::
+   There appears to be varying pin-out configurations on different modules - beware!
+
+.. note::
+
+  * If you're already using the listed GPIO pins for Data/Command and/or Reset,
+    you can select other pins and pass :py:attr:`gpio_DC` and/or :py:attr:`gpio_RST`
+    argument specifying the new *GPIO* pin numbers in your serial interface create
+    call (this applies to PCD8544, ST7567 and ST7735).
+
+  * Because CE is connected to CE0, the display is available on SPI port 0. You
+    can connect it to CE1 to have it available on port 1. If so, pass
+    :py:attr:`port=1` in your serial interface create call.
+
+The GPIO pins used for this SPI connection are the same for all versions of the
+Raspberry Pi, up to and including the Raspberry Pi 3 B.
```

Comment:
It's probably safe to change this to 4.

Reply:
Same comment as above but I did make the change for this one.